### PR TITLE
bump awssdk version for a bugfix on http conn leakage

### PR DESCRIPTION
**Pull request by klsince**
`bugfix`

The latest aws sdk version 2.20.83 includes a critical bugfix https://github.com/aws/aws-sdk-java-v2/pull/4066 to avoid http conn leakage when aws requests, like S3 getObject, get interrupted.

### CODE REVIEW

**Review by walterddr — COMMENTED** [review]
lgtm. qq: can't we extract the aws.sdk.version somewhere less verbose?

**Review by ege-st — APPROVED** [review]

**Comment by klsince** [comment]
> can't we extract the aws.sdk.version somewhere less verbose

I'll let the PR pass tests and then polish

**Review by walterddr — APPROVED** [review]

**Comment by walterddr** [comment]
error seems related to the s3mock package with the latest aws lib:
```
 Tests run: 17, Failures: 1, Errors: 0, Skipped: 16, Time elapsed: 7.534 s <<< FAILURE! - in org.apache.pinot.plugin.filesystem.S3PinotFSTest
Error:  org.apache.pinot.plugin.filesystem.S3PinotFSTest.setUp  Time elapsed: 7.309 s  <<< FAILURE!
software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 405, Request ID: null)
```

File: pinot-plugins/pinot-file-system/pinot-s3/pom.xml [hunk 1]
```diff
@@ -183,27 +183,9 @@
     </dependency>
     <dependency>
       <groupId>com.adobe.testing</groupId>
-      <artifactId>s3mock-testng</artifactId>
+      <artifactId>s3mock-testcontainers</artifactId>
```
Comment (klsince) [inline pinot-plugins/pinot-file-system/pinot-s3/pom.xml:186]: `s3mock-testcontainers` is [recommended](https://github.com/adobe/S3Mock#s3mock-java) as it avoids bring in deps from s3mock transitively, which brings in the spring framework and also requires some [specific keystore](https://github.com/adobe/S3Mock/issues/720). 

The new version of s3mock is more accurate on handling the leading and trailing slash in s3 object names, so the changes on the unit tests as made in this PR.

**Review by gortiz — APPROVED** [review]

**Pull request merged**